src/App.tsx

```diff
@@ -15,36 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+  None = '',
+}
+
 export const App: React.FC = () => {
+  const [sortedGoods, setSortedGoods] = useState<string[]>([
+    ...goodsFromServer,
+  ]);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const sortGoods = (goods: string[], type: SortType): string[] => {
+    const sorted = [...goods];
+
+    switch (type) {
+      case SortType.Alphabetical:
+        sorted.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.Length:
+        sorted.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+
+    if (isReversed) {
+      sorted.reverse();
+    }
+
+    return sorted;
+  };
+
+  const handleSort = (type: SortType) => {
+    const newSortType = type || sortType;
+
+    setSortType(newSortType);
+
+    const sorted = sortGoods(goodsFromServer, newSortType);
+
+    setSortedGoods(sorted);
+  };
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
+    setSortedGoods(prevGoods => [...prevGoods].reverse());
+  };
+
+  const resetSort = () => {
+    setSortType(SortType.None);
+    setIsReversed(false);
+    setSortedGoods([...goodsFromServer]);
+  };
+
+  const buttonStyle = (type: SortType) => (sortType === type ? '' : 'is-light');
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', buttonStyle(SortType.Alphabetical))}
```

Comment:
Consider marking the active button to explain to the user what action is active. For instance, you could apply a different style or class to the currently active sort type button. [UX]